docs/en/CSMM/functions.md

```diff
@@ -0,0 +1,55 @@
+---
+title: CSMM Functions
+---
+
+# Functions
+
+Functions are actions you can do inside the execution of a custom command.
+
+### wait(x)
+
+Will wait x seconds before continuing with command execution. Warning: you should not use this for long delays. Whenever CSMM restarts, it will forget about commands that are still running.
+
+Example:
+```
+pm \${player.steamId} "You will be teleported in one minute."; wait(60); cpm-mvw \${player.steamId} home
+```
+
+### addCurrency(playerId, amount)
+
+With this function, you can add or subtract currency from a players balance.
+
+Examples:
+```
+pm \${player.steamId} "Thanks for your vote!"; addCurrency(\${player.id}, 50)
+
+addCurrency(\${player.id}, -500); teleh \${player.entityId}
+```
+
+### setRole(steamId, role)
+
+Sets a players role. The role parameter is the name of the role, this has to be **exactly** the same. This can be used if you have an \$adddonor command, in the vote reward command to give voters access to special commands, etc.
+
+Example:
+```
+setRole(\${player.steamId}, Donator)
+```
+
+### sendDiscord(channel, message)
+
+Sends a discord message via the CSMM bot. Note that you must have already authorised the bot in your Discord server.
+
+To identify a channel, send a message within your Discord server using the format `\\#channel-name`. The message will appear in the chat window with the channel ID rather than its display name. Copy this number (excluding the <# and >) to set the channel you wish your Discord message to appear in.
```

Comment:
Another option of getting the IDs is to enable developer mode on Discord and use the right click menu. We could add a link to the Discord docs for this: https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-

Idk which method is 'easier' though :smile:

Reply:
I tried this method too but it feels like it involves a lot more steps, and it's a more varied process between the channel / user / role methods. I can write up both tho, if you like :)